src/auth/auth.service.js

```diff
@@ -0,0 +1,55 @@
+import { Injectable, Dependencies } from '@nestjs/common';
+import { ConfigService } from '../config/config.service';
+import crypto from 'crypto';
+import moment from 'moment';
+
+@Injectable()
+@Dependencies(ConfigService)
+export class AuthService {
+  constructor(configService) {
+    this.configService = configService;
+  }
+
+  async authIdentity(appId, timestamp, signature) {
+    if (appId !== this.configService.auth.appId) {
+      return null;
+    }
+
+    if (
+      signature !==
+      crypto
+        .createHmac(
+          this.configService.auth.hmacAlgorithm,
+          this.configService.auth.appSecret,
+        )
+        .update(timestamp)
+        .digest('hex')
+    ) {
+      return null;
+    }
+
+    if (
```

Comment:
why you have this design (not before or after AUTH_TIMESTAMP_TOLERANCE_IN_MS)?

Reply:
It's mentioned in the [authentication design decisions](https://github.com/MiffyLiye/test-backend/blob/master/docs/README.md#authentication-design-decisions) section, so that the service can decide when the auth headers should expire, and  the decrypted information will not pass authentication in the future since the timestamp is expired.